example/src/ParticipantView.tsx

```diff
@@ -1,38 +1,43 @@
 import * as React from 'react';
 
 import { Image, StyleSheet, ViewStyle } from 'react-native';
-import type { Participant } from 'livekit-client';
-import { useParticipant, VideoView } from '@livekit/react-native';
+import {
+  isTrackReference,
+  TrackReferenceOrPlaceholder,
+  useEnsureTrackRef,
+  useIsMuted,
+  useIsSpeaking,
+  useParticipantInfo,
+  VideoTrack,
+} from '@livekit/react-native';
 import { View } from 'react-native';
 import { Text } from 'react-native';
 import { useTheme } from '@react-navigation/native';
 export type Props = {
-  participant: Participant;
+  trackRef: TrackReferenceOrPlaceholder;
   style?: ViewStyle;
   zOrder?: number;
   mirror?: boolean;
 };
 export const ParticipantView = ({
   style = {},
-  participant,
+  trackRef,
   zOrder,
   mirror,
 }: Props) => {
-  const { cameraPublication, screenSharePublication } =
-    useParticipant(participant);
-  let videoPublication = cameraPublication ?? screenSharePublication;
-
+  const trackReference = useEnsureTrackRef(trackRef);
```

Comment:
if `trackRef` is required as prop, we can skip `useEnsureTrackRef`. This hook only tries to either grab trackRef from props or, if not available, read it from the context.